functional-tests/test-bed/src/test/java/org/arquillian/smart/testing/ftest/configurationfile/ConfigurationFilePerModuleTest.java

```diff
@@ -0,0 +1,71 @@
+package org.arquillian.smart.testing.ftest.configurationfile;
+
+import java.util.Collection;
+import org.arquillian.smart.testing.configuration.Configuration;
+import org.arquillian.smart.testing.ftest.testbed.configuration.builder.ConfigurationBuilder;
+import org.arquillian.smart.testing.ftest.testbed.project.Project;
+import org.arquillian.smart.testing.ftest.testbed.project.TestResults;
+import org.arquillian.smart.testing.ftest.testbed.testresults.TestResult;
+import org.arquillian.smart.testing.rules.TestBed;
+import org.arquillian.smart.testing.rules.git.GitClone;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.arquillian.smart.testing.RunMode.SELECTING;
+import static org.arquillian.smart.testing.ftest.testbed.TestRepository.testRepository;
+import static org.arquillian.smart.testing.ftest.testbed.configuration.Strategy.CHANGED;
+import static org.arquillian.smart.testing.ftest.testbed.configuration.Strategy.NEW;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ConfigurationFilePerModuleTest {
+    @ClassRule
+    public static final GitClone GIT_CLONE = new GitClone(testRepository());
+
+    @Rule
+    public final TestBed testBed = new TestBed(GIT_CLONE);
+
+    @Test
+    public void should_load_configuration_from_module_if_it_has_config_file() {
+        // given
+        final Project project = testBed.getProject();
+
+        final Configuration parentConfiguration = new ConfigurationBuilder()
+                .strategies(NEW, CHANGED)
+                .scm()
+                    .lastChanges("2")
+                .build()
+            .build();
+
+        final Configuration newConfiguration = new ConfigurationBuilder()
+                .strategies(NEW)
+                .mode(SELECTING)
+            .build();
+
+        final Configuration changedConfiguration = new ConfigurationBuilder()
+                .strategies(CHANGED)
+                .mode(SELECTING)
+            .build();
+
+        project.configureSmartTesting()
+                .withConfiguration(parentConfiguration)
+                    .withConfigFile()
+                .withConfiguration(newConfiguration)
+                    .withConfigFileIn("config/impl-base")
+                .withConfiguration(changedConfiguration)
+                    .withConfigFileIn("junit/core")
```

Comment:
This chain is a bit hard for me to follow. Can you explain what is the desired flow of action here? I think `with...` makes it a bit blurry to follow as not everything should be prefixed like that.